tasks/notify.py

```diff
@@ -171,3 +177,68 @@ def print_gitlab_ci(
 
     # Print
     print_gitlab_ci_configuration(yml, sort_jobs=sort)
+
+
+@task
+def gitlab_ci_diff(ctx, before: str | None = None, after: str | None = None, pr_comment: bool = False):
+    """
+    Creates a diff from two gitlab-ci configurations.
+
+    - before: Git ref without new changes, None for LCA commit between HEAD and main
+    - after: Git ref with new changes, None for current local configuration
+    - pr_comment: If True, post the diff as a comment in the PR
+    - NOTE: This requires a full api token access level to the repository
+    """
+
+    pr_comment_head = 'Gitlab CI Configuration Changes'
+    if pr_comment:
+        job_url = os.environ['CI_JOB_URL']
+
+    try:
+        before_name = before or "merge base"
+        after_name = after or "local files"
+        before = before or ctx.run(f'git merge-base {DEFAULT_BRANCH} HEAD', hide=True).stdout.strip()
+
+        print(f'Getting after changes config ({color_message(after_name, Color.BOLD)})')
+        after_config = get_gitlab_ci_configuration(ctx, git_ref=after)
+
+        print(f'Getting before changes config ({color_message(before_name, Color.BOLD)})')
+        before_config = get_gitlab_ci_configuration(ctx, git_ref=before)
+
+        diff = GitlabCIDiff(before_config, after_config)
+
+        if not diff:
+            print(color_message("No changes in the gitlab-ci configuration", Color.GREEN))
+
+            # Remove comment if no changes
+            if pr_comment:
+                pr_commenter(ctx, pr_comment_head, delete=True, force_delete=True)
+
+            return
+
+        # Display diff
+        print('\nGitlab CI configuration diff:')
+        with gitlab_section('Gitlab CI configuration diff'):
+            print(diff.display(cli=True))
+
+        if pr_comment:
+            print('\nSending / updating PR comment')
+            comment = diff.display(cli=False)
+            try:
+                pr_commenter(ctx, pr_comment_head, comment)
+            except Exception:
+                print(color_message('Warning: Failed to send full diff, sending only job link', Color.ORANGE))
+
+                pr_commenter(ctx, pr_comment_head, f'Too many changes, see the [job log]({job_url}) for details')
```

Comment:
Is that really accurate? I feel like the error could be related to other things than `Too many changes`